.github/workflows/staging-release.yaml

```diff
@@ -1,121 +1,125 @@
-    # Just sync everything to the release server
     steps:
-    - name: Hashed known hosts value
-      id: known_hosts
+    - name: Checkout code
+      uses: actions/checkout@v2
+
+    - name: Setup runner
       run: |
-          OUTPUT=$(ssh-keyscan -H ${{ secrets.FLUENTBITIO_HOST }})
-          echo ::set-output name=OUTPUT::$OUTPUT
+        sudo apt-get install debsigs createrepo aptly rsync
+      shell: bash
 
-    - name: Install SSH Key
-      uses: shimataro/ssh-key-action@v2
+    - name: Import GPG key for signing
+      id: import_gpg
+      uses: crazy-max/ghaction-import-gpg@v4
       with:
-        key: ${{ secrets.FLUENTBITIO_SSHKEY }}
-        known_hosts: ${{ steps.known_hosts.outputs.OUTPUT }}
+        gpg_private_key: ${{ secrets.GPG_PRIVATE_KEY }}
 
-    - name: Sync packages from staging on S3
+    # Download the current release bucket
+    # Add everything from staging
+    # Sign and set up metadata for it all
+    # Upload to release bucket
+
+    - name: Sync packages from buckets on S3
       run: |
-        rm -rf ./releases/
-        mkdir -p ./releases/
+        mkdir -p packaging/releases
+        aws s3 sync "s3://${{ secrets.AWS_S3_BUCKET_RELEASE }}" packaging/releases/ --no-progress
+        aws s3 sync "s3://${{ secrets.AWS_S3_BUCKET_STAGING }}" packaging/releases/ --no-progress
+      env:
+        AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+        AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+      shell: bash
 
-        if [ -n "${AWS_S3_ENDPOINT}" ]; then
-          ENDPOINT="--endpoint-url ${AWS_S3_ENDPOINT}"
-        fi
-        aws s3 sync "s3://${{ secrets.AWS_S3_BUCKET_STAGING }}" ./releases/ --no-progress ${ENDPOINT}
+    - name: GPG set up keys for signing
+      run: |
+        gpg --export -a "${{ steps.import_gpg.outputs.name }}" > packaging/releases/fluentbit.key
+        rpm --import packaging/releases/fluentbit.key
+      shell: bash
 
-        rsync -avz ./releases/ $USERNAME@$HOST:/home/$USERNAME/apt
+    - name: Update repo info
+      run: |
+        VERSION=$(cat packaging/releases/latest-version.txt)
+        packaging/update-repos.sh "$VERSION" packaging/releases/
+        rm -f packaging/releases/latest-version.txt
       env:
-        HOST: ${{ secrets.FLUENTBITIO_HOST }}
-        USERNAME: ${{ secrets.FLUENTBITIO_USERNAME }}
-        KEY: ${{ secrets.FLUENTBITIO_SSHKEY }}
+        GPG_KEY: ${{ steps.import_gpg.outputs.name }}
       shell: bash
-      working-directory: packaging
 
-  staging-release-packages-s3:
-    runs-on: ubuntu-18.04 # no createrepo on more recent runners
+    - name: Sync to release bucket on S3
+      run: |
+        aws s3 sync packaging/releases/ "s3://${{ secrets.AWS_S3_BUCKET_RELEASE }}" --follow-symlinks --no-progress
+      env:
+        AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+        AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+      shell: bash
+
+  # We have two options here:
```

Comment:
@niedbalski what's your take here? Option 2 is closer to the current process but needs more manual work to complete. Option 1 provides a sync of the full release bucket but as it grows you will need to filter out the RPM/Debs for the version just built to add or do what you need on the manual server.